test/services/courses/services/coursePermission.test.js

```diff
@@ -0,0 +1,219 @@
+const { expect } = require('chai');
+const app = require('../../../../src/app');
+const testObjects = require('../../helpers/testObjects')(app);
+
+const coursePermissionService = app.service('/courses/:scopeId/userPermissions');
+
+describe('PermissionService', async () => {
+	const studentPermissions = [
+		'BASE_VIEW',
+		'DASHBOARD_VIEW',
+		'TOOL_VIEW',
+		'PASSWORD_EDIT',
+		'FOLDER_DELETE',
+		'FOLDER_CREATE',
+		'FILE_DELETE',
+		'FILE_CREATE',
+		'FILE_MOVE',
+		'ACCOUNT_EDIT',
+		'CALENDAR_VIEW',
+		'CALENDAR_EDIT',
+		'CALENDAR_CREATE',
+		'FEDERALSTATE_VIEW',
+		'HELPDESK_CREATE',
+		'TOPIC_VIEW',
+		'LINK_CREATE',
+		'NEWS_VIEW',
+		'NOTIFICATION_VIEW',
+		'NOTIFICATION_EDIT',
+		'NOTIFICATION_CREATE',
+		'PWRECOVERY_VIEW',
+		'PWRECOVERY_EDIT',
+		'PWRECOVERY_CREATE',
+		'RELEASES_VIEW',
+		'ROLE_VIEW',
+		'USERGROUP_VIEW',
+		'COURSEGROUP_CREATE',
+		'COURSEGROUP_EDIT',
+		'SYSTEM_VIEW',
+		'USER_VIEW',
+		'USER_EDIT',
+		'HOMEWORK_VIEW',
+		'COMMENTS_VIEW',
+		'COMMENTS_CREATE',
+		'COMMENTS_EDIT',
+		'SUBMISSIONS_VIEW',
+		'SUBMISSIONS_CREATE',
+		'SUBMISSIONS_EDIT',
+		'FILESTORAGE_VIEW',
+		'FILESTORAGE_EDIT',
+		'FILESTORAGE_CREATE',
+		'CONTENT_VIEW',
+		'CONTENT_NON_OER_VIEW',
+		'FILESTORAGE_REMOVE',
+		'TEAM_VIEW',
+		'TEAM_EDIT',
+		'TEAM_CREATE',
+	];
+
+	const teacherPermissions = [
+		'ACCOUNT_CREATE',
+		'COURSE_EDIT',
+		'HOMEWORK_EDIT',
+		'HOMEWORK_CREATE',
+		'LESSONS_VIEW',
+		'NEWS_CREATE',
+		'NEWS_EDIT',
+		'SCHOOL_NEWS_EDIT',
+		'STUDENT_CREATE',
+		'STUDENT_DELETE',
+		'STUDENT_SKIP_REGISTRATION',
+		'SUBMISSIONS_SCHOOL_VIEW',
+		'TEACHER_CREATE',
+		'TOOL_CREATE',
+		'TOOL_EDIT',
+		'TOOL_NEW_VIEW',
+		'TOPIC_CREATE',
+		'TOPIC_EDIT',
+		'USER_CREATE',
+		'USERGROUP_CREATE',
+		'USERGROUP_EDIT',
+		'TEAM_INVITE_EXTERNAL',
+		'BASE_VIEW',
+		'DASHBOARD_VIEW',
+		'TOOL_VIEW',
+		'PASSWORD_EDIT',
+		'FOLDER_DELETE',
+		'FOLDER_CREATE',
+		'FILE_DELETE',
+		'FILE_CREATE',
+		'FILE_MOVE',
+		'ACCOUNT_EDIT',
+		'CALENDAR_VIEW',
+		'CALENDAR_EDIT',
+		'CALENDAR_CREATE',
+		'FEDERALSTATE_VIEW',
+		'HELPDESK_CREATE',
+		'TOPIC_VIEW',
+		'LINK_CREATE',
+		'NEWS_VIEW',
+		'NOTIFICATION_VIEW',
+		'NOTIFICATION_EDIT',
+		'NOTIFICATION_CREATE',
+		'PWRECOVERY_VIEW',
+		'PWRECOVERY_EDIT',
+		'PWRECOVERY_CREATE',
+		'RELEASES_VIEW',
+		'ROLE_VIEW',
+		'USERGROUP_VIEW',
+		'COURSEGROUP_CREATE',
+		'COURSEGROUP_EDIT',
+		'SYSTEM_VIEW',
+		'USER_VIEW',
+		'USER_EDIT',
+		'HOMEWORK_VIEW',
+		'COMMENTS_VIEW',
+		'COMMENTS_CREATE',
+		'COMMENTS_EDIT',
+		'SUBMISSIONS_VIEW',
+		'SUBMISSIONS_CREATE',
+		'SUBMISSIONS_EDIT',
+		'FILESTORAGE_VIEW',
+		'FILESTORAGE_EDIT',
+		'FILESTORAGE_CREATE',
+		'CONTENT_VIEW',
+		'CONTENT_NON_OER_VIEW',
+		'FILESTORAGE_REMOVE',
+		'TEAM_VIEW',
+		'TEAM_EDIT',
+		'TEAM_CREATE',
```

Comment:
pro tip: use `test/helpers/services/roles` to create two test roles with only a few unique permissions, use these to create two users (`test/helpers/services/users`), and add them to your course 🎉